feat(curriculum): add workshop python functions (#68279)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-kitchen-inventory-tracker/6a3e524fc3394ef52ed5b797.md

---
id: 6a3e524fc3394ef52ed5b797
title: Step 3
challengeType: 20
dashedName: step-3
---

# --description--

Functions have their own "local scope". This means variables created inside a function are like secret ingredients—they can only be accessed inside that specific function.

Inside your `take_inventory` function, replace the `pass` keyword with a new variable named `total_items` and assign it the sum of `eggs`, `flour`, and `sugar`.

# --hints--

You should define a variable named `total_items` inside your `take_inventory` function.

```js
({
  test: () => assert(
    runPython(`_Node(_code).find_function("take_inventory").find_assignments("total_items")`)
  )
})
```

Your total_items variable should be the sum of eggs, flour, and sugar.

```js
({
  test: () => assert(
    runPython(`

>       _Node(_code).find_function("take_inventory").find_assignments("total_items")[0].is_equivalent("eggs + flour + sugar")

Reviewer: ```suggestion
    variable = _Node(_code).find_function("take_inventory").find_variable("total_items")
    variable.is_equivalent("total_items = eggs + flour + sugar") or variable.is_equivalent("total_items = eggs + sugar + flour") or ...
```

please make sure all 6 permutations are allowed

---
id: 6a3e524fc3394ef52ed5b797
title: Step 3
challengeType: 20
dashedName: step-3
---

# --description--

> Functions have their own "local scope". This means variables created inside a function are like secret ingredients—they can only be accessed inside that specific function.

Reviewer: ```suggestion
Functions have their own "local scope". This means variables created inside a function are like secret ingredients: they can only be accessed inside that specific function.
```

Inside your `take_inventory` function, replace the `pass` keyword with a new variable named `total_items` and assign it the sum of `eggs`, `flour`, and `sugar`.

# --hints--

You should define a variable named `total_items` inside your `take_inventory` function.

```js
({
  test: () => assert(
    runPython(`_Node(_code).find_function("take_inventory").has_variable("total_items")`)
  )
})
```

> Your total_items variable should be the sum of eggs, flour, and sugar.

Reviewer: ```suggestion
Your `total_items` variable should be the sum of `eggs`, `flour`, and `sugar`.
```

---
id: 6a3e524fc3394ef52ed5b797
title: Step 3
challengeType: 20
dashedName: step-3
---

# --description--

> Functions have their own "local scope". This means variables created inside a function are like secret ingredients: they can only be accessed inside that specific function.

Reviewer 2: "secret ingredients" implies something hidden/special, but local scope is really about accessibility restriction, not secrecy. You can try the below alternative or if you find some better example, do let us know.

```suggestion
Functions have their own "local scope". This means variables created inside a function are like ingredients kept in that kitchen's pantry: they can only be accessed inside that specific function.
```

> Notice how your function can read both the local variables defined inside the function and the global variables defined outside the function!

Reviewer 2: I believe this should come after the next paragraph. You are talking about reading a local variable here, but the code is not reading any local variable; instead, it is creating one and initializing it with the sum of the ingredients.

> However, functions are not completely isolated. They can read "global" variables defined outside their scope and use them for various tasks—like performing calculations, making decisions based on their values, or combining them to create entirely new local variables.

Reviewer: ```suggestion
However, functions are not completely isolated. They can read "global" variables defined outside their scope. They can use those variables to perform calculations, make decisions, or create new local variables.
```
